src/go/package/app/services/tag.go

```diff
@@ -0,0 +1,52 @@
+package services
+
+import (
+	"github.com/goravel/framework/facades"
+
+	"market.goravel.dev/package/app/models"
+	protobase "market.goravel.dev/proto/base"
+)
+
+type Tag interface {
+	GetTags(packageID, name string, pagination *protobase.Pagination, total *int64) ([]*models.Tag, error)
+}
+
+type TagImpl struct {
+	tagModel models.TagInterface
+}
+
+func NewTagImpl() *TagImpl {
+	return &TagImpl{
+		tagModel: models.NewTag(),
+	}
+}
+
+func (r *TagImpl) GetTags(packageID, name string, pagination *protobase.Pagination, total *int64) ([]*models.Tag, error) {
+	var tags []*models.Tag
+	query := facades.Orm().Query()
+
+	if packageID != "" {
+		query = query.Join("JOIN package_tags ON package_tags.tag_id = tags.id").
```

Comment:
Don't need to use the join sql, if packageID != "", we can get  the tagIDs from the package_tags through packageID: split to two sqls. `Json` is inefficiency, split is better.